pkg/util/email/email.go

```diff
@@ -0,0 +1,40 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package email
+
+import (
+	"regexp"
+
+	// Not using recommended package "errors" for the need of Errorf()
+	"github.com/pkg/errors"
+)
+
+// Validate validates that a string is in a valid email format
+func Validate(email string) error {
+	emailRegexp := regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+\\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
+
+	if len(email) > 254 {
+		return errors.New("an email address must not exceed 254 characters")
+	}
+
+	if !emailRegexp.MatchString(email) {
+		return errors.Errorf("%v is not a valid email address format", email)
```

Comment:
We can actually change this to use `fmt.Errorf`. Not sure why (I think I did) "github.com/pkg/errors" would be preferred over of `fmt`